org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceAwareList.java

```diff
@@ -0,0 +1,215 @@
+package org.osgi.test.junit5.service;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+import org.osgi.test.common.service.ServiceAware;
+
+public class ServiceAwareList<E> implements List<E> {
```

Comment:
You could generalize this in to a `ListSupplierDelegate`.

```java
public class ListSupplierDelegate<E> implements List<E> {
    private final Supplier<? extends List<? extends E>> supplier;
    public ListSupplierDelegate(Supplier<? extends List<? extends E>> supplier) {
        this.supplier = Objects.requireNonNull(supplier);
    }
    @Override
	public void forEach(Consumer<? super E> action) {
		supplier.get()
			.forEach(action);
	}
```
 You create one with `new ListSupplierDelegate<>(configuration::getServices)`.

Reply:
good idea